R/getNeatOrder.R

```diff
@@ -17,38 +17,64 @@
+#' 
+#' @details 
+#' It's important to note that the sechm package does actually have the functionality for plotting a heatmap using this radial theta angle ordering, though only by using an MDS ordination. 
+#' 
+#' This functionality can be found at:
+#' 
+#'  \url{https://bioconductor.org/packages/3.18/bioc/vignettes/sechm/inst/doc/sechm.html#row-ordering}. 
+#'  
+#' That being said, the \code{getNeatOrder} function is more modular and separate to the plotting, and 
+#' can be applied to any kind of ordinated data which can be valuable depending on the use case.
+#' 
+#' @references
+#' The below paper outlines the NeatMap method in more detail:
+#' 
+#' Rajaram, S., Oono, Y. NeatMap - non-clustering heat map alternatives in R. BMC Bioinformatics 11, 45 (2010). https://doi.org/10.1186/1471-2105-11-45.
+#' 
+#' It can be found at:
+#' 
+#'  \url{https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-11-45}.
 #' 
 #' @name getNeatOrder
 #' 
 #' @examples
-#' # Load required libraries
+#' ## Load the required libraries and dataset
 #' library(mia)
-#' 
-#' # Load the dataset
+#' library(scater)
+#' library(sechm)
 #' data(peerj13075)
 #' 
-#' # Agglomerate by Order and transform the data
-#' tse_order <- agglomerateByRank(peerj13075, rank = "order", onRankOnly = TRUE)
-#' tse_order <- transformAssay(tse_order, assay.type = "counts", method="relabundance", MARGIN = "samples", name="relabundance")
-#' tse_order <- transformAssay(tse_order, assay.type="relabundance", method="z", MARGIN = "features", name="z")
-#' z_transformed_data <- assay(tse_order, "z")
-#' 
-#' # Get the top taxa and perform PCA
-#' top_taxa <- getTopFeatures(tse_order, top = 10, assay.type="z")
-#' top_feature_data <- z_transformed_data[top_taxa, ]
-#' pca_results <- prcomp(top_feature_data, scale = TRUE)
-#' scores_pca <- pca_results$x[, 1:2]
-#' 
-#' # Sort by radial theta and subset the transformed data
-#' sorted_order <- getNeatOrder(scores_pca, dimensions = c(1, 2), centering_method = "mean")
-#' ordered_transformed_data <- z_transformed_data[sorted_order, ]
+#' ## Group data by taxonomic order
+#' tse <- agglomerateByRank(peerj13075, rank = "order", onRankOnly = TRUE)
+#' 
+#' ## Transform the samples into relative abundances
+#' tse <- transformAssay(tse, assay.type = "counts", method="relabundance", MARGIN = "samples", name="relabundance")
+#' 
+#' ## Transform the features (taxa) into zero mean, unit variance (z transformation)
+#' tse <- transformAssay(tse, assay.type="relabundance", method="z", MARGIN = "features", name="z")
+#' 
+#' ## Perform PCA using calculatePCA
+#' pca_result <- calculatePCA(tse, ncomponents = 10, assay.type = "z")
+#' 
+#' ## Add PCA results to the TreeSE object
+#' reducedDim(tse, "PCA") <- pca_result
+#' 
+#' ## Sort by radial theta and sort the original assay data
+#' sorted_order <- getNeatOrder(reducedDim(tse, "PCA"), dimensions = c(1, 2), centering.method = "mean")
+#' tse <- tse[, sorted_order]
```

Comment:
The use of this with heatmap is not demonstrated, whereas sechm is. 

It would be better to show heatmap with getNeatOrder+heatmap, or both that one and sechm (for comparison). But in function examples you are supposed to primarily  demonstrate how that function (getNeatOrder) is used to support heatmap plotting.